bot/core.py

```diff
@@ -64,6 +66,15 @@ async def on_message(self, message: discord.Message):
         app = get_app(message)
         logger.debug(f"App: {app}")
         if app is not None:
+            self.analytics.record_command(
+                app=message.content.split()[0].strip(PREFIX),
+                full_command=message.content,
+                discord_user_id=message.author.id,
+                discord_server_id=message.guild.id,
+                server_name=read_config_str(
+                    message.guild.id, ConfigKeys.GAME_SERVER, ""
+                ),
+            )
             await app.run()
```

Comment:
How will this capture execution time if it runs before the command runs?

Reply:
Eventual next step would be to refactor the run method to return boolean on success, log error message if applicable, and then log execution time.  Doesn't seem critical for first pass bc of the amount of refactoring, but leaving the columns in the table for future proofing.